src/cargo/util/toml/mod.rs

```diff
@@ -1595,7 +1595,42 @@ pub struct InheritableFields {
     ws_root: PathBuf,
 }
 
+/// Defines simple getter methods for inheritable fields.
+macro_rules! inheritable_field_getter {
+    ( $(($path:literal, $field:ident -> $ret:ty),)* ) => (
+        $(
+            #[doc = concat!("Gets the field `workspace.", $path, "`.")]
+            pub fn $field(&self) -> CargoResult<$ret> {
+                if let Some(val) = &self.$field {
+                    return Ok(val.clone())
+                }
+                bail!("`workspace.{}` was not defined", $path);
+            }
+        )*
+    )
+}
+
 impl InheritableFields {
+    inheritable_field_getter! {
+        ("dependencies",          dependencies  -> BTreeMap<String, TomlDependency>),
```

Comment:
Nit: Add a comment about keeping this ordered, similar to [this comment](https://github.com/rust-lang/cargo/blob/5febbe5587b74108165f748e79a4f8badbdf5e0e/src/cargo/core/features.rs#L719)
```suggestion
        // Please keep this list lexicographically ordered.
        ("dependencies",          dependencies  -> BTreeMap<String, TomlDependency>),
```